### DISCUSSION TITLE
How do I create a bilingual project?

### DISCUSSION THREAD

**Discussion by Josh596** [WeblateOrg/weblate#10683] [Q&A]
Hey everyone, 
Sorry about the long post. Any help is appreciated. 

I generate Qt Linguist .ts files, and after reading through the weblate documentation, and various discussions on the issues section of this repo, I believed that my .ts files were bilingual. 

I run weblate v5.2 via docker.

Steps I follow
1. Create a weblate project
2. Create a project component by using the "Translate document" option

## When I try to create a bilingual component as described in "https://docs.weblate.org/en/latest/formats/qt.html"
<img width="945" alt="Screenshot 2023-12-22 at 17 55 49" src="https://github.com/WeblateOrg/weblate/assets/64589157/89500390-0741-4635-8b27-8dbd2a442111">
<img width="942" alt="Screenshot 2023-12-22 at 17 56 03" src="https://github.com/WeblateOrg/weblate/assets/64589157/e8671765-24f7-4b0c-9dba-5b9e2bb64c5a">
<img width="899" alt="Screenshot 2023-12-22 at 17 56 36" src="https://github.com/WeblateOrg/weblate/assets/64589157/5fa8193b-ea9c-4353-a380-2b238a4530d9">

<img width="889" alt="Screenshot 2023-12-22 at 17 56 47" src="https://github.com/WeblateOrg/weblate/assets/64589157/06df8d5c-bc3e-42b1-9a80-e1a5c6965a78">

After I create the component, 0 source strings are discovered
<img width="945" alt="Screenshot 2023-12-22 at 17 57 22" src="https://github.com/WeblateOrg/weblate/assets/64589157/e77808d6-f139-4df1-ac0e-8f599ae8e602">

## When I try to create a monolingual component as described in "https://docs.weblate.org/en/latest/formats/qt.html"
My source string is all messed up, and I experience an issue already discussed [here](https://github.com/WeblateOrg/weblate/issues/1962)
<img width="928" alt="Screenshot 2023-12-22 at 18 10 12" src="https://github.com/WeblateOrg/weblate/assets/64589157/949c239a-96c5-4043-924b-8182c91359da">

<img width="905" alt="Screenshot 2023-12-22 at 18 09 56" src="https://github.com/WeblateOrg/weblate/assets/64589157/6de0bc99-d020-45ce-95c4-20dc4cf113f6">

Here is a snippet from the `app_en.ts` file I use here
```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1" language="en" sourcelanguage="en">
<context>
    <name>BottomSection</name>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="45"/>
        <source>Unsharp Mask</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="48"/>
        <source>Auto Adjustment</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="51"/>
        <source>Denoise</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="54"/>
        <source>Comic</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="57"/>
        <source>Color Correction</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="61"/>
        <source>Automatic Sharpen</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="64"/>
        <source>Automatic Sharpen 2</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../widgets/controlBar/pages/page_1/bottomSection.py" line="67"/>
        <source>Remove Material</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>BrightnessControl</name>
    <message>
        <location filename="../widgets/controlBar/controls/brightness.py" line="7"/>
        <source>Brightness</source>
        <translation type="unfinished"></translation>
    </message>
</TS>

```

**Comment by nijel** [comment]
You currently can not create a bilingual component via document upload (see https://github.com/WeblateOrg/weblate/issues/11085). Upload the file in a zip file, and you will be able to configure everything.

I think there is an existing issue for this...

**Reply by Josh596** [reply]
Thank you for your response. I tried adding the translation file(app_en.ts) to a zip file and using the "Upload Translation Files" option when creating my components. Unfortunately, when I try to configure it as a bilingual component as described [here](https://docs.weblate.org/en/latest/formats/qt.html) it still does not discover any source string. 

Any help would be appreciated.

**Reply by nijel** [reply]
Do you get some error, ok the translation is just blank? Can you share that file and configuration?
